controllers/keda/hpa.go

```diff
@@ -134,6 +136,16 @@ func (r *ScaledObjectReconciler) newHPAForScaledObject(ctx context.Context, logg
 		return nil, err
 	}
 
+	// store hpaName in the ScaledObject
```

Comment:
Let's move this into `createAndDeployNewHPA()` once there is successful HPA creation. This function should just return HPA object.